packages/odie-client/src/components/send-message-input/index.tsx

```diff
@@ -22,9 +22,27 @@ export const OdieSendMessageButton = ( {
 	const { trackEvent, chatStatus, shouldUseHelpCenterExperience } = useOdieAssistantContext();
 	const sendMessage = useSendChatMessage();
 	const shouldBeDisabled = chatStatus === 'loading' || chatStatus === 'sending';
+	const [ isMessageSizeValid, setIsMessageSizeValid ] = useState( true );
+
+	const isMessageExceedingMaxLength = ( messageLength: number ) => {
+		// zendesk api validation
+		return messageLength <= 4096;
+	};
+
+	const onKeyUp = useCallback( () => {
+		// Only triggered when the message is empty
+		// used to remove validation message.
+		setIsMessageSizeValid( true );
+	}, [] );
 
 	const sendMessageHandler = useCallback( async () => {
-		if ( inputRef.current?.value.trim() === '' || shouldBeDisabled ) {
+		const message = inputRef.current?.value.trim();
+		const messageLength = message?.length || 0;
+		const isMessageLengthValid = isMessageExceedingMaxLength( messageLength );
+
+		setIsMessageSizeValid( isMessageLengthValid );
+
+		if ( message === '' || shouldBeDisabled || ! isMessageLengthValid ) {
```

Comment:
```suggestion
		if ( message === '' || shouldBeDisabled || ( shouldUseHelpCenterExperience && ! isMessageLengthValid ) ) {
```

Do we want to only apply this to our flag for now?

Reply:
Addressed 👍